webapp/lib/controller.dart

```diff
@@ -267,48 +277,86 @@ bool isProjectStale(model.NeedsReplyData projectData) {
   }
 }
 
-void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
-  projectTimers[selectedProject]?.cancel();
+String getWatchdogTimer(Object data) {
+  if (data is model.NeedsReplyData) {
+    return  selectedProject;
+  } else if (data is model.SystemMetricsData) {
+    return SYSTEM_METRICS_ROOT_COLLECTION_KEY;
+  } else {
+    throw new model.ModelDoesNotExistException('Model does not exist');
+  }
+}
+
+void setupWatchdogTimer<T>(T latestData, [bool stale = false]) {
```

Comment:
How are the generics helping here?

Reply:
I've seen they're really not that necessary. Refactored them out.